### DISCUSSION TITLE
how to intergate langium server with glsp and vscode

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1231] [Q&A]
there is a requirement that need to transform industrial Programming Language LD(Ladder Diagram) to other specific lauguage like C++ or PLC ST (Structured Text）language,when user drag a component or delete a component ,corresponding text is changed to,so i am considering use glsp+langium+vscode extension+vscode webview,now i have start glsp server with vscode extension(using workflow glsp example as template).i know crossmodel project intergate langium with glsp,but not  intergate with vscode extension,so how should i  intergate langium with both glsp and vscode,can i regard langium server as a part of glsp server?

**Comment by msujew** [comment]
@zrwsmd You might be better off asking the kind people over at the glsp project. While they have some experience working with Langium, we don't have any experience with glsp.
